### ISSUE TITLE
loki plugin grows without end

### ISSUE THREAD

**Issue by badsmoke** [grafana/loki#1868]
**Describe the bug**
the loki plugin keeps growing despite daemon.json settings until the disk is full

**To Reproduce**
1. Started Loki (1.3.0)
2. loki plugin from 04.12.2019
3.

**Expected behavior**
the loki plugin file should have a maximum size of 100gb, but it will be significantly larger without end.

**Environment:**
80 hardware systems with ubuntu 18.04 installed, from each of these systems will send three container logs to loki.
on this systems the disk is also regularly filled by the loki plugin

```
root@loki-server:/mnt/docker# docker plugin ls
ID                  NAME                DESCRIPTION           ENABLED
b081d3bdcd40        loki:latest         Loki Logging Driver   true
root@loki-server:/mnt/docker# du --max-depth=1 -h ./
16K     ./builder
64K     ./buildkit
12M     ./containers
15M     ./image
188K    ./network
412G    ./plugins    !!!!
0       ./swarm
0       ./trust
52M     ./volumes
0       ./docke
48G     ./btrfs
0       ./tmp
0       ./runtimes
459G    ./
root@loki-server:/mnt/docker# cat /etc/docker/daemon.json
{
  "experimental" : true,
  "live-restore": true,
  "log-driver": "loki",
    "log-opts": {
        "loki-url": "http://localhost/logs/api/prom/push",
        "loki-batch-size": "400",
        "max-size": "100g",
        "max-file": "2"
      }
}


```

**Comment by cyriltovena** [comment]
What is taking this space inside this folder ?

**Comment by cyriltovena** [comment]
I think those settings are per container. So 200g max per container. You might want to reduce that down to 1g per container max even that this would be a lot if you have many containers on the same host.

**Comment by badsmoke** [comment]
sorry i forgot the size of the log files.
one log file was about 300gb, so it was already three times as big as set, even if it is only per container

**Comment by ushuz** [comment]
@badsmoke Did you set log driver for the container in question with `docker run` or compose file? It might not inherit the options from `daemon.json`.

**Comment by badsmoke** [comment]
all containers were started by docker-compose, it also shows that the loki driver is used by default. i have also restart the computer and docker daemon and also several times  docker-compose down/up

**Comment by cyriltovena** [comment]
We could definitely add a log to avoid confusion when the driver start.

**Comment by badsmoke** [comment]
this solution with "no log file" would already solve the problem but it would limit the work if there is no possibility to view the logs directly on the host. It also does not explain why the set limits "max-file", "max-size" have no effect.

**Comment by ushuz** [comment]
> It also does not explain why the set limits "max-file", "max-size" have no effect.

@badsmoke For me `max-file` `max-size` work just well (docker 18.06.2 + loki 1.3.0). Look at code, the options are properly passed along. So I guess your case may be related to the docker's json-log component, which handles the actual file creation and rotation underlay, but is outside the scope of loki.

**Comment by floriandotorg** [comment]
Having the exact same problem in 2025. loki docker plugin logs growing indefinitely even though max-file and max-size are set.  Any solutions?

**Comment by floriandotorg** [comment]
Having the exact same problem in 2025. loki docker plugin logs growing indefinitely even though max-file and max-size are set. Any solutions?

**Comment by erickythierry** [comment]
I had the same problem with Loki taking up the entire system disk, preventing everything else from working properly.
The only solution I found was to simply delete the files inside:
`/var/lib/docker/plugins/[plugin-id]/rootfs/var/log/docker`
Then I restarted the server.
This freed up more than half of the disk.

I'm going to create a cron process to do this monthly since there doesn't seem to be a practical solution.
